tests/php/Model/SubmittedFileFieldTest.php

```diff
@@ -61,4 +66,33 @@ public function testDeletingSubmissionRemovesFile()
 
         $this->assertNull($liveVersion, 'Live file has been deleted');
     }
+
+    public function testGetFormattedValue()
+    {
+        $fileName = $this->submittedFile->getFileName();
+
+        $this->file->CanViewType = 'OnlyTheseUsers';
+        $this->file->write();
+        
+        $this->loginWithPermission('ADMIN');
+        $this->assertEquals(
+            sprintf(
+                '%s - <a href="http://localhost/assets/3c01bdbb26/test-SubmittedFileFieldTest.txt" target="_blank">Download File</a>',
+                $fileName
+            ),
+            $this->submittedFile->getFormattedValue()->value
+        );
+            
+        $this->loginWithPermission('CMS_ACCESS_CMSMain');
+        $this->assertEquals(
+            sprintf(
+                '<span class="icon font-icon-lock">%s - <i>You don\'t have the right permissions to download this file</i></span>',
+                $fileName
+            ),
+            $this->submittedFile->getFormattedValue()->value
+        );
+
+        $store = Injector::inst()->get(AssetStore::class);
+        $this->assertFalse($store->canView($fileName, $this->file->getHash()));
```

Comment:
More of a nice to have, but I find that adding descriptive messages to assertion makes what is being tested clearer.

```suggestion
        $this->assertFalse(
            $store->canView($fileName, $this->file->getHash()),
            'Users without canView rights on the file should not have been session granted access to it'
        );
```

Reply:
DONE